src/psygnal/_group.py

```diff
@@ -183,12 +192,23 @@ def disconnect(self, slot: Callable | None = None, missing_ok: bool = True) -> N
         super().disconnect(slot, missing_ok)
 
 
+class SignalGroupMetaclass(type):
```

Comment:
can you accomplish this without metaclasses?  mypyc doesn't support metaclasses, and in general I've been happy with using mypyc, and wouldn't want to drop it just for this